frontend/src/components/ProfileBookingHistory.js

```diff
@@ -1,64 +1,120 @@
-import React from "react"
-import { CardContent, Stack, Card, Typography } from "@mui/material"
+import React, { useState, useEffect } from "react"
+import { db } from "../firebase"
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Card,
+  CardContent,
+  Typography,
+  Stack,
+} from "@mui/material"
+import {
+  doc,
+  collection,
+  where,
+  query,
+  orderBy,
+  getDocs,
+} from "@firebase/firestore"
+
+const ProfileBookingHistory = ({ userId }) => {
+  const [bookings, setBookings] = useState([])
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    if (!userId) return
+    setLoading(true)
+
+    const fetchData = async () => {
+      try {
+        const userDocRef = doc(db, "users", userId)
+        const q = query(
+          collection(db, "bookings"),
+          where("user_id", "==", userDocRef),
+          orderBy("check_in", "desc")
+        )
+
+        const querySnapshot = await getDocs(q)
+
+        const fetchedBookings = querySnapshot.docs.map((doc) => ({
+          id: doc.id,
+          ...doc.data(),
+        }))
+        setBookings(fetchedBookings)
+      } catch (error) {
+        console.error("Error fetching user bookings: ", error)
+      } finally {
+        setLoading(false)
+      }
+    }
```

Comment:
It seems like the bookings data is being fetch here within the component. The booking data is already fetched in profile and could be used instead.